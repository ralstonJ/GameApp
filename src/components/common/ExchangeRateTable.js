import React from 'react';

const renderRates = exchangeRates =>
  Object.keys(exchangeRates).map((data, i) => {
    return (
      <tr key={i}>
        <td>{data}</td>
        <td> {exchangeRates[data]}</td>
      </tr>
    );
  });
const ExchangeRateTable = ({ exchangeRates }) => {
  return (
    <div>
      <table>
        <tbody>{renderRates(exchangeRates)}</tbody>
      </table>
    </div>
  );
};

export default ExchangeRateTable;
