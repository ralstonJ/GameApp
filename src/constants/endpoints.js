export const LATEST_EXCHANGE_RATES = 'https://api.exchangeratesapi.io/latest';
