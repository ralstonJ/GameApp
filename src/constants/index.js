export const exchangeRates = { CAD: 1.723, USD: 1.89 };

export const ERROR_MESSAGE = 'Data fetch fails';
