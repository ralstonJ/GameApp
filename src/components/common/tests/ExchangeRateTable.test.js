import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { exchangeRates } from '../../../constants';
import ExchangeRateTable from '../ExchangeRateTable';

afterEach(cleanup);
describe('ExchangeRateTable Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <ExchangeRateTable exchangeRates={exchangeRates} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('to have 2 rates', async () => {
    const { container } = render(
      <ExchangeRateTable exchangeRates={exchangeRates} />
    );
    const tr = container.querySelectorAll('tr');
    expect(tr).toHaveLength(2);
  });
});
