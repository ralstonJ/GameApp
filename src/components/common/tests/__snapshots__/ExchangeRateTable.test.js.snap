// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRateTable Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <table>
      <tbody>
        <tr>
          <td>
            CAD
          </td>
          <td>
             1.723
          </td>
        </tr>
        <tr>
          <td>
            USD
          </td>
          <td>
             1.89
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
